From 5c463e47654cda2e0525d97c550363f30d81984e Mon Sep 17 00:00:00 2001
From: md_5 <git@md-5.net>
Date: Tue, 8 Jul 2014 09:01:50 +1000
Subject: [PATCH] Disable ResourceLeakDetector

Hopefully no plugins are somehow leaking buffers, but disabling the ResourceLeakDetector is a good thing for performance of the Netty IO subsytem (it's been disabled in BungeeCord for a while now).

diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index bacb595..3d0a91d 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -115,6 +115,7 @@ public abstract class MinecraftServer implements ICommandListener, Runnable, IMo
     // Spigot end
 
     public MinecraftServer(OptionSet options, Proxy proxy) { // CraftBukkit - signature file -> OptionSet
+        net.minecraft.util.io.netty.util.ResourceLeakDetector.setEnabled( false ); // Spigot - disable
         this.X = new UserCache(this, a);
         j = this;
         this.d = proxy;
-- 
1.9.1

